### PR TITLE
release/public-v2: bit-for-bit reproducibility for regional runs when changing MPI decomposition

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

These PRs solve the b4b reproducibility issues for regional runs when changing the MPI layout (decomposition) as described in https://github.com/ufs-community/ufs-weather-model/issues/288.

With the changes in https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/68, results are b4b identical in DEBUG, REPRO and PROD mode when the dycore is compiled in 64bit, regardless of the physics suite.

When the dycore is compiled din 32bit, results are b4b identical in DEBUG, REPRO and PROD mode for the RRFSv1alpha suite (or, more generally, any suite that does not use GFDL-MP with saturation adjustment = fast physics turned on). Results are b4b identical in DEBUG and REPRO mode (but not in PROD mode) for the GFSv15p2 suite (which uses GFDL-MP with saturation adjustment = fast physics turned on). 

Additional regression tests were added to `tests/rt.conf` to provide the necessary test coverage going forward. Note that for release/public-v2, `32BIT=Y` is the default and thus note explicitly spelled out in `rt.conf`.

Also: added orion.intel to the tier-1 platforms for release/public-v2.

### Issue(s) addressed

These PRs solve the "b4b reproducibility for regional runs on the ESG grid when changing the MPI decomposition" of issue https://github.com/ufs-community/ufs-weather-model/issues/288.

## Testing

Regression tests passed on all tier-1 platforms for release/public-v2, including the newly added orion.intel.

## Dependencies

https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/68
https://github.com/NOAA-EMC/fv3atm/pull/245
https://github.com/ufs-community/ufs-weather-model/pull/409

### CODE REVIEW

**Review by junwang-noaa — APPROVED** [review]

**Comment by climbfuji** [comment]
Thanks @junwang-noaa ! I'll wait until jet is back, rerun the tests there and merge afterwards.

**Pull request merged**
